ehealthid-cli/src/main/java/com/oviva/ehealthid/esgen/FedRegistrationCommand.java

```diff
@@ -0,0 +1,183 @@
+package com.oviva.ehealthid.esgen;
+
+import com.nimbusds.jose.jwk.Curve;
+import com.nimbusds.jose.jwk.JWK;
+import com.nimbusds.jose.jwk.JWKSet;
+import com.nimbusds.jose.jwk.KeyType;
+import com.nimbusds.jose.jwk.KeyUse;
+import com.oviva.ehealthid.esgen.RegistratonFormRenderer.Model;
+import com.oviva.ehealthid.esgen.RegistratonFormRenderer.Model.Environment;
+import com.oviva.ehealthid.esgen.RegistratonFormRenderer.Model.Scope;
+import com.oviva.ehealthid.fedclient.api.EntityStatementJWS;
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse.BodyHandlers;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.stream.Stream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import picocli.CommandLine;
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Option;
+
+@Command(
+    name = "fedreg",
+    mixinStandardHelpOptions = true,
+    description =
+        """
+        Generate requests for registration in the eHealthID federation. Sent to Gematik.
+        """)
+public class FedRegistrationCommand implements Callable<Integer> {
+
+  private static final Logger logger = LoggerFactory.getLogger(FedRegistrationCommand.class);
+
+  @Option(
+      names = {"-e", "--environment"},
+      description =
+          "the environment to register for, either TU (Testumgebung) or RU (Referenzumgebung)",
+      defaultValue = "TU",
+      required = true)
+  private Environment environment;
+
+  @Option(
+      names = {"-m", "--member-id"},
+      description = "the member ID to register for, needs to be requested from Gematik",
+      required = true)
+  private String memberId;
+
+  @Option(
+      names = {"-i", "--iss", "--issuer-uri"},
+      description = "the issuer uri of the 'Fachdienst' identiy provider",
+      required = true)
+  private URI issuerUri;
+
+  public static void main(String[] args) {
+
+    int exitCode = 1;
+    try {
+      var main = new FedRegistrationCommand();
+      exitCode = new CommandLine(main).execute(args);
+    } catch (Exception e) {
+      logger.atError().setCause(e).log("key generator failed");
+      System.exit(1);
+    }
+    System.exit(exitCode);
+  }
+
+  public Integer call() throws Exception { // your business logic goes here...
+
+    var entityConfiguration = fetchEntityConfiguration();
+    for (var key : entityConfiguration.jwks().getKeys()) {
+      validateKey(key, KeyUse.SIGNATURE);
+    }
+
+    writeRegistrationForm(entityConfiguration);
+    return 0;
+  }
+
+  private void writeRegistrationForm(EntityConfiguration entityConfiguration) throws IOException {
```

Comment:
personally I would delegate this function and subsequent private methods to a separate class

Reply:
I need class scopes values, while I can move it out it would be unnecessarily complex in my opinion.